indexing-service/src/main/java/org/apache/druid/indexing/overlord/http/OverlordResource.java

```diff
@@ -227,6 +227,34 @@ public Response isLeader()
     }
   }
 
+  @GET
+  @Path("/lockedIntervals")
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response getTaskLockedIntervals(@Context HttpServletRequest request)
+  {
+    // Perform authorization check
+    final ResourceAction resourceAction = new ResourceAction(
```

Comment:
Any particular this API needs a finer grained access control rather than using the StateResourceFilter ?
The StateResourceFilter is already used for API like /taskStatus which I think is similar in access control to this API.

Reply:
Fixed.